File: VB/AppleAccounts.vb
Imports Microsoft.VisualBasic
Imports Microsoft.VisualBasic.CompilerServices
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Diagnostics
Imports System.Drawing
Imports System.Linq
Imports System.Runtime.CompilerServices
Imports System.Windows.Forms

Namespace SuperImpactor
	<DesignerGenerated>
	Public Class AppleAccounts
		Inherits Form
		Private components As IContainer

		Friend Overridable Property cmdClose As Button
			Get
				stackVariable1 = Me._cmdClose
				Return stackVariable1
			End Get
			<MethodImpl(MethodImplOptions.Synchronized)>
			Set(ByVal value As System.Windows.Forms.Button)
				Dim eventHandler As System.EventHandler = New System.EventHandler(AddressOf Me.cmdClose_Click)
				Dim button As System.Windows.Forms.Button = Me._cmdClose
				If (button IsNot Nothing) Then
					RemoveHandler button.Click,  eventHandler
				End If
				Me._cmdClose = value
				button = Me._cmdClose
				If (button IsNot Nothing) Then
					AddHandler button.Click,  eventHandler
				End If
			End Set
		End Property

		Friend Overridable Property cmdRemove As Button
			Get
				stackVariable1 = Me._cmdRemove
				Return stackVariable1
			End Get
			<MethodImpl(MethodImplOptions.Synchronized)>
			Set(ByVal value As System.Windows.Forms.Button)
				Dim eventHandler As System.EventHandler = New System.EventHandler(AddressOf Me.cmdRemove_Click)
				Dim button As System.Windows.Forms.Button = Me._cmdRemove
				If (button IsNot Nothing) Then
					RemoveHandler button.Click,  eventHandler
				End If
				Me._cmdRemove = value
				button = Me._cmdRemove
				If (button IsNot Nothing) Then
					AddHandler button.Click,  eventHandler
				End If
			End Set
		End Property

		Friend Overridable Property cmdRemoveAll As Button
			Get
				stackVariable1 = Me._cmdRemoveAll
				Return stackVariable1
			End Get
			<MethodImpl(MethodImplOptions.Synchronized)>
			Set(ByVal value As System.Windows.Forms.Button)
				Dim eventHandler As System.EventHandler = New System.EventHandler(AddressOf Me.cmdRemoveAll_Click)
				Dim button As System.Windows.Forms.Button = Me._cmdRemoveAll
				If (button IsNot Nothing) Then
					RemoveHandler button.Click,  eventHandler
				End If
				Me._cmdRemoveAll = value
				button = Me._cmdRemoveAll
				If (button IsNot Nothing) Then
					AddHandler button.Click,  eventHandler
				End If
			End Set
		End Property

		Friend Overridable Property Label1 As Label

		Friend Overridable Property lstCydia As ListView

		Public Sub New()
			MyBase.New()
			AddHandler MyBase.Load,  New EventHandler(AddressOf Me.AppleAccounts_Load)
			Me.InitializeComponent()
		End Sub

		Private Sub AppleAccounts_Load(ByVal sender As Object, ByVal e As EventArgs)
			Me.LoadAccounts()
		End Sub

		Private Sub cmdClose_Click(ByVal sender As Object, ByVal e As EventArgs)
			MyBase.Close()
		End Sub

		Private Sub cmdRemove_Click(ByVal sender As Object, ByVal e As EventArgs)
			If (Me.lstCydia.SelectedItems.Count <= 0) Then
				Interaction.MsgBox("Select account to remove", MsgBoxStyle.OkOnly, Nothing)
			Else
				Dim item As ListViewItem = Me.lstCydia.SelectedItems(0)
				Database.DeleteAccount(item.SubItems(0).Text)
				Me.LoadAccounts()
			End If
		End Sub

		Private Sub cmdRemoveAll_Click(ByVal sender As Object, ByVal e As EventArgs)
			Database.DeleteAccounts()
			Me.lstCydia.Clear()
		End Sub

		<DebuggerNonUserCode>
		Protected Overrides Sub Dispose(ByVal disposing As Boolean)
			Try
				If (If(Not disposing, False, Me.components IsNot Nothing)) Then
					Me.components.Dispose()
				End If
			Finally
				MyBase.Dispose(disposing)
			End Try
		End Sub

		<DebuggerStepThrough>
		Private Sub InitializeComponent()
			Me.cmdClose = New Button()
			Me.cmdRemove = New Button()
			Me.cmdRemoveAll = New Button()
			Me.Label1 = New Label()
			Me.lstCydia = New ListView()
			MyBase.SuspendLayout()
			Me.cmdClose.Location = New Point(262, 357)
			Me.cmdClose.Name = "cmdClose"
			Me.cmdClose.Size = New System.Drawing.Size(104, 25)
			Me.cmdClose.TabIndex = 15
			Me.cmdClose.Text = "Close"
			Me.cmdClose.UseVisualStyleBackColor = True
			Me.cmdRemove.Location = New Point(262, 24)
			Me.cmdRemove.Name = "cmdRemove"
			Me.cmdRemove.Size = New System.Drawing.Size(104, 25)
			Me.cmdRemove.TabIndex = 14
			Me.cmdRemove.Text = "Remove"
			Me.cmdRemove.UseVisualStyleBackColor = True
			Me.cmdRemoveAll.Location = New Point(262, 57)
			Me.cmdRemoveAll.Name = "cmdRemoveAll"
			Me.cmdRemoveAll.Size = New System.Drawing.Size(104, 25)
			Me.cmdRemoveAll.TabIndex = 13
			Me.cmdRemoveAll.Text = "Remove All"
			Me.cmdRemoveAll.UseVisualStyleBackColor = True
			Me.Label1.AutoSize = True
			Me.Label1.Location = New Point(12, 5)
			Me.Label1.Name = "Label1"
			Me.Label1.Size = New System.Drawing.Size(117, 13)
			Me.Label1.TabIndex = 12
			Me.Label1.Text = "List of Stored Accounts"
			Me.lstCydia.FullRowSelect = True
			Me.lstCydia.GridLines = True
			Me.lstCydia.HideSelection = False
			Me.lstCydia.Location = New Point(12, 24)
			Me.lstCydia.Name = "lstCydia"
			Me.lstCydia.Size = New System.Drawing.Size(239, 358)
			Me.lstCydia.TabIndex = 11
			Me.lstCydia.UseCompatibleStateImageBehavior = False
			MyBase.AutoScaleDimensions = New SizeF(6!, 13!)
			MyBase.AutoScaleMode = System.Windows.Forms.AutoScaleMode.Font
			MyBase.ClientSize = New System.Drawing.Size(377, 393)
			MyBase.ControlBox = False
			MyBase.Controls.Add(Me.cmdClose)
			MyBase.Controls.Add(Me.cmdRemove)
			MyBase.Controls.Add(Me.cmdRemoveAll)
			MyBase.Controls.Add(Me.Label1)
			MyBase.Controls.Add(Me.lstCydia)
			MyBase.FormBorderStyle = System.Windows.Forms.FormBorderStyle.FixedSingle
			MyBase.MaximizeBox = False
			MyBase.MinimizeBox = False
			MyBase.Name = "AppleAccounts"
			Me.Text = "Stored Apple Accounts"
			MyBase.ResumeLayout(False)
			MyBase.PerformLayout()
		End Sub

		Private Sub LoadAccounts()
			Dim accounts As Dictionary(Of String, String) = Database.GetAccounts()
			Me.lstCydia.Clear()
			Me.lstCydia.View = View.Details
			Me.lstCydia.Columns.Add("Apple Id")
			Me.lstCydia.AutoResizeColumns(ColumnHeaderAutoResizeStyle.ColumnContent)
			Me.lstCydia.AutoResizeColumns(ColumnHeaderAutoResizeStyle.HeaderSize)
			Dim count As Integer = accounts.Keys.Count - 1
			For i As Integer = 0 To count Step 1
				Dim listViewItem As System.Windows.Forms.ListViewItem = New System.Windows.Forms.ListViewItem() With
				{
					.Text = accounts.Keys.ElementAt(i)
				}
				listViewItem.SubItems.Add(accounts.Keys.ElementAt(i))
				Me.lstCydia.Items.Add(listViewItem)
			Next

		End Sub
	End Class
End Namespace

File: super-impactor source/C#/AppleAccounts.cs
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Diagnostics;
using System.Drawing;
using System.Linq;
using System.Runtime.CompilerServices;
using System.Windows.Forms;

namespace SuperImpactor
{
	[DesignerGenerated]
	public class AppleAccounts : Form
	{
		private IContainer components;

		internal virtual Button cmdClose
		{
			get
			{
				stackVariable1 = this._cmdClose;
				return stackVariable1;
			}
			[MethodImpl(MethodImplOptions.Synchronized)]
			set
			{
				EventHandler eventHandler = new EventHandler(this.cmdClose_Click);
				Button button = this._cmdClose;
				if (button != null)
				{
					button.Click -= eventHandler;
				}
				this._cmdClose = value;
				button = this._cmdClose;
				if (button != null)
				{
					button.Click += eventHandler;
				}
			}
		}

		internal virtual Button cmdRemove
		{
			get
			{
				stackVariable1 = this._cmdRemove;
				return stackVariable1;
			}
			[MethodImpl(MethodImplOptions.Synchronized)]
			set
			{
				EventHandler eventHandler = new EventHandler(this.cmdRemove_Click);
				Button button = this._cmdRemove;
				if (button != null)
				{
					button.Click -= eventHandler;
				}
				this._cmdRemove = value;
				button = this._cmdRemove;
				if (button != null)
				{
					button.Click += eventHandler;
				}
			}
		}

		internal virtual Button cmdRemoveAll
		{
			get
			{
				stackVariable1 = this._cmdRemoveAll;
				return stackVariable1;
			}
			[MethodImpl(MethodImplOptions.Synchronized)]
			set
			{
				EventHandler eventHandler = new EventHandler(this.cmdRemoveAll_Click);
				Button button = this._cmdRemoveAll;
				if (button != null)
				{
					button.Click -= eventHandler;
				}
				this._cmdRemoveAll = value;
				button = this._cmdRemoveAll;
				if (button != null)
				{
					button.Click += eventHandler;
				}
			}
		}

		internal virtual Label Label1
		{
			get;
			[MethodImpl(MethodImplOptions.Synchronized)]
			set;
		}

		internal virtual ListView lstCydia
		{
			get;
			[MethodImpl(MethodImplOptions.Synchronized)]
			set;
		}

		public AppleAccounts()
		{
			base.Load += new EventHandler(this.AppleAccounts_Load);
			this.InitializeComponent();
		}

		private void AppleAccounts_Load(object sender, EventArgs e)
		{
			this.LoadAccounts();
		}

		private void cmdClose_Click(object sender, EventArgs e)
		{
			base.Close();
		}

		private void cmdRemove_Click(object sender, EventArgs e)
		{
			if (this.lstCydia.SelectedItems.Count <= 0)
			{
				Interaction.MsgBox("Select account to remove", MsgBoxStyle.OkOnly, null);
			}
			else
			{
				ListViewItem item = this.lstCydia.SelectedItems[0];
				Database.DeleteAccount(item.SubItems[0].Text);
				this.LoadAccounts();
			}
		}

		private void cmdRemoveAll_Click(object sender, EventArgs e)
		{
			Database.DeleteAccounts();
			this.lstCydia.Clear();
		}

		[DebuggerNonUserCode]
		protected override void Dispose(bool disposing)
		{
			try
			{
				if ((!disposing ? false : this.components != null))
				{
					this.components.Dispose();
				}
			}
			finally
			{
				base.Dispose(disposing);
			}
		}

		[DebuggerStepThrough]
		private void InitializeComponent()
		{
			this.cmdClose = new Button();
			this.cmdRemove = new Button();
			this.cmdRemoveAll = new Button();
			this.Label1 = new Label();
			this.lstCydia = new ListView();
			base.SuspendLayout();
			this.cmdClose.Location = new Point(262, 357);
			this.cmdClose.Name = "cmdClose";
			this.cmdClose.Size = new System.Drawing.Size(104, 25);
			this.cmdClose.TabIndex = 15;
			this.cmdClose.Text = "Close";
			this.cmdClose.UseVisualStyleBackColor = true;
			this.cmdRemove.Location = new Point(262, 24);
			this.cmdRemove.Name = "cmdRemove";
			this.cmdRemove.Size = new System.Drawing.Size(104, 25);
			this.cmdRemove.TabIndex = 14;
			this.cmdRemove.Text = "Remove";
			this.cmdRemove.UseVisualStyleBackColor = true;
			this.cmdRemoveAll.Location = new Point(262, 57);
			this.cmdRemoveAll.Name = "cmdRemoveAll";
			this.cmdRemoveAll.Size = new System.Drawing.Size(104, 25);
			this.cmdRemoveAll.TabIndex = 13;
			this.cmdRemoveAll.Text = "Remove All";
			this.cmdRemoveAll.UseVisualStyleBackColor = true;
			this.Label1.AutoSize = true;
			this.Label1.Location = new Point(12, 5);
			this.Label1.Name = "Label1";
			this.Label1.Size = new System.Drawing.Size(117, 13);
			this.Label1.TabIndex = 12;
			this.Label1.Text = "List of Stored Accounts";
			this.lstCydia.FullRowSelect = true;
			this.lstCydia.GridLines = true;
			this.lstCydia.HideSelection = false;
			this.lstCydia.Location = new Point(12, 24);
			this.lstCydia.Name = "lstCydia";
			this.lstCydia.Size = new System.Drawing.Size(239, 358);
			this.lstCydia.TabIndex = 11;
			this.lstCydia.UseCompatibleStateImageBehavior = false;
			base.AutoScaleDimensions = new SizeF(6f, 13f);
			base.AutoScaleMode = System.Windows.Forms.AutoScaleMode.Font;
			base.ClientSize = new System.Drawing.Size(377, 393);
			base.ControlBox = false;
			base.Controls.Add(this.cmdClose);
			base.Controls.Add(this.cmdRemove);
			base.Controls.Add(this.cmdRemoveAll);
			base.Controls.Add(this.Label1);
			base.Controls.Add(this.lstCydia);
			base.FormBorderStyle = System.Windows.Forms.FormBorderStyle.FixedSingle;
			base.MaximizeBox = false;
			base.MinimizeBox = false;
			base.Name = "AppleAccounts";
			this.Text = "Stored Apple Accounts";
			base.ResumeLayout(false);
			base.PerformLayout();
		}

		private void LoadAccounts()
		{
			Dictionary<string, string> accounts = Database.GetAccounts();
			this.lstCydia.Clear();
			this.lstCydia.View = View.Details;
			this.lstCydia.Columns.Add("Apple Id");
			this.lstCydia.AutoResizeColumns(ColumnHeaderAutoResizeStyle.ColumnContent);
			this.lstCydia.AutoResizeColumns(ColumnHeaderAutoResizeStyle.HeaderSize);
			int count = checked(accounts.Keys.Count - 1);
			for (int i = 0; i <= count; i = checked(i + 1))
			{
				ListViewItem listViewItem = new ListViewItem()
				{
					Text = accounts.Keys.ElementAt<string>(i)
				};
				listViewItem.SubItems.Add(accounts.Keys.ElementAt<string>(i));
				this.lstCydia.Items.Add(listViewItem);
			}
		}
	}
}